src/components.rs

```diff
@@ -77,3 +78,17 @@ pub struct LineOfSight {
     /// The limit of the unit vision.
     pub range: i32,
 }
+
+/// Order queue component for micro-control (and maybe bots)
+pub struct OrderQueue {
+    /// List of orders for a unit.
+    // TODO: Use vecdeque to optimize. But currently this might be considered overoptimization.
+    pub orders: Vec<UnitOrder>,
```

Comment:
VecDeque is actually the correct structure to use here ^u^. Gives O(1) instead of O(n), so its definitely worth it.